### PR TITLE
New studies

### PR DESCRIPTION
**Pull request by janvanrijn**

### CODE REVIEW

File: openml_OS/views/pages/api_new/v1/xml/study-attach-detach.tpl.php [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<oml:study_<?php echo $function_type; ?> xmlns:oml="http://openml.org/openml">
+  <oml:id><?php echo $id; ?></oml:id>
+  <oml:main_knowledge_type><?php echo $main_knowledge_type; ?></oml:main_knowledge_type>
```
Comment (joaquinvanschoren) [inline openml_OS/views/pages/api_new/v1/xml/study-attach-detach.tpl.php]: You called this an entity type earlier, and now a knowledge type?
Why 'main'? A study has 1 type, right?

File: openml_OS/views/pages/frontend/new/post.php [hunk 2]
```diff
@@ -222,7 +222,7 @@
   $name = $this->input->post( 'study_title' );
   $alias = $this->input->post( 'study_alias' );
   $description = $this->input->post( 'description' );
-  $sid = $this->Study->create( $name, $alias, $description, $user_id );
+  $sid = $this->Study->legacy_create( $name, $alias, $description, $user_id );
```
Comment (joaquinvanschoren) [inline openml_OS/views/pages/frontend/new/post.php]: Should we take this down from the frontend? Or put a notice that they will be discontinued in their current form?

File: openml_OS/models/api/v1/Api_study.php [hunk 3]
```diff
@@ -41,20 +59,194 @@ function bootstrap($format, $segments, $request_type, $user_id) {
...
+      $xml = simplexml_load_file($description['tmp_name']);
+    }
+    
+    $study = all_tags_from_xml($xml->children('oml', true), $this->xml_fields_study);
+    
+    if (!in_array($study['main_entity_type'], $legal_entity_types)) {
+      $this->returnError(1033, $this->version);
+      return;
+    }
+    $link_entities = $this->_get_linked_entities_from_xml($xml, $legal_entity_types);
```
Comment (joaquinvanschoren) [inline openml_OS/models/api/v1/Api_study.php]: What are entities here? The IDs of the things we want to the study?

File: openml_OS/views/pages/api_new/v1/xsd/openml.study.upload.xsd [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+</xs:simpleType>
+<xs:simpleType name="basic_latin4096">
+  <xs:restriction base="xs:string">
+    <xs:pattern value="\p{IsBasicLatin}*"/>
+    <xs:maxLength value="4096" />
+    <xs:minLength value="1" />
+  </xs:restriction>
+</xs:simpleType>
+
+<xs:complexType name="data">
```
Comment (joaquinvanschoren) [inline openml_OS/views/pages/api_new/v1/xsd/openml.study.upload.xsd]: Isn't it overkill to have separate XSD types for what are basically just IDs? Why not let all of these be normal lists of integers?

File: openml_OS/views/pages/frontend/new/post.php (hunk 2 above)
Comment (janvanrijn) [reply openml_OS/views/pages/frontend/new/post.php]: Yes, take it down from the frontend

File: openml_OS/views/pages/api_new/v1/xml/study-attach-detach.tpl.php (hunk 1 above)
Comment (janvanrijn) [reply openml_OS/views/pages/api_new/v1/xml/study-attach-detach.tpl.php]: done

File: openml_OS/views/pages/frontend/new/post.php (hunk 2 above)
Comment (janvanrijn) [reply openml_OS/views/pages/frontend/new/post.php]: take it down, would be great

File: openml_OS/views/pages/api_new/v1/xsd/openml.study.upload.xsd (hunk 4 above)
Comment (janvanrijn) [reply openml_OS/views/pages/api_new/v1/xsd/openml.study.upload.xsd]: it's because of 

```
<entity type>
  <entity id></entity id>
<entity type>
```

**Pull request merged**